src/DistributedLock.Etcd/EtcdLeaseDistributedLockHandle.cs

```diff
@@ -0,0 +1,44 @@
+using dotnet_etcd.interfaces;
+using Etcdserverpb;
+using Medallion.Threading.Internal;
+
+namespace Medallion.Threading.Etcd;
+
+public class EtcdLeaseDistributedLockHandle : IDistributedSynchronizationHandle
+{
+    public string Key;
+
+    internal EtcdLeaseDistributedLockHandle(string key, EtcdClientWrapper client, long leaseKey)
+    {
+        this.Key = key ?? throw new ArgumentNullException(nameof(key));
+        this.Client = client ?? throw new ArgumentNullException(nameof(client));
+        this.LeaseKey = leaseKey;
+        // Because this is a lease, managed finalization mostly won't be strictly necessary here. Where it comes in handy is:
+        // (1) Ensuring blob deletion if we own the blob
+        // (2) Helping release infinite-duration leases (rare case)
+        // (3) In testing, avoiding having to wait 15+ seconds for lease expiration
+    }
+
+    private long LeaseKey { get; set; }
+
+    private EtcdClientWrapper Client { get; set; }
+
+    /// <summary>
+    /// Implements <see cref="IDistributedSynchronizationHandle.HandleLostToken"/>
+    /// </summary>
+    public CancellationToken HandleLostToken => new();
```

Comment:
See previous comment; we may be able to support this. If no, we should return `CancellationToken.None;` to be explicit